src/simplewallet/simplewallet.cpp

```diff
@@ -1462,6 +1464,72 @@ bool simple_wallet::handle_command_line(const boost::program_options::variables_
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::prompt_for_new_node()
+{
+    bool connect = false;
+    std::string node_url;
+
+    message_writer() << tr("Do you want to connect to a remote node?");
+    std::string confirm = command_line::input_line(tr("(Y/Yes/N/No): "));
+    if(std::cin.eof())
+    {
+      LOG_ERROR("Unexpected std::cin.eof() - Exited simple_wallet::prompt_for_new_node()");
+      return false;
+    }
+    connect = command_line::is_yes(confirm);
+
+    if (connect) {
+        message_writer() << tr("Please enter remote node: ");
+        node_url = command_line::input_line(tr("http(s)://[domain]:[port] (default port 18081): "));
+
+        boost::regex rgx("^(.*)://([A-Za-z0-9\\-\\.]+)(:[0-9]+)?(.*)");
```

Comment:
http[s]:// ought to be optional. Does hostname make sense to be optional ? What is the trailing (.*) for ?

Reply:
So what would be the best approach? Disregard the https and make it default to it or will there be nodes providing non-SSL encrypted nodes?

Good point, `(.*)` is not really needed, as it would contain the path in an URI.

Reply:
You don't need to disregard https/http, just make it optional (ie, someone can enter something typical like 1.2.3.4:18081 and it'll be fine). There may be nodes using SSL at some point, but AFAIK there is no support for this currently.